File: app/models/user_log.py
# app/models/user_log.py
from sqlalchemy import Column, Integer, String, Float, DateTime
from app.db.base import Base
from datetime import datetime


class UserLog(Base):
    __tablename__ = "user_logs"

    id = Column(Integer, primary_key=True, index=True)
    method = Column(String, nullable=False)
    path = Column(String, nullable=False)
    status_code = Column(Integer, nullable=False)
    duration = Column(Float, nullable=False)
    timestamp = Column(DateTime, default=datetime.utcnow)

    ip = Column(String, nullable=True)
    user_agent = Column(String, nullable=True)

File: express-app/db/models/user_log.js
// express-app/db/models/user_log.js
module.exports = (sequelize, DataTypes) => {
  const UserLog = sequelize.define('UserLog', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
      indexed: true,
    },
    method: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    path: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    status_code: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    duration: {
      type: DataTypes.FLOAT,
      allowNull: false,
    },
    timestamp: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
    },
    ip: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    user_agent: {
      type: DataTypes.STRING,
      allowNull: true,
    },
  }, {
    tableName: 'user_logs',
  });

  return UserLog;
};